framework/include/actions/ActionWarehouse.h

```diff
@@ -148,6 +148,28 @@ class ActionWarehouse : public ConsoleStreamInterface
     return action_vector;
   }
 
+  /**
+   * Retrieve an action on a specific task with its type.
+   * @param name The task name.
+   * @return The action pointer. Null means that such an action does not exist.
+   */
+  template <class T>
+  const T * getActionByTask(const std::string & task)
```

Comment:
Right, I was not thinking something like `AddMaterialAction`, although the use case in this PR always only have one action. Without this helper function, the same functionality has to be repeated several places, like checking the existence of a task, loop through actions on the task. How about I have a check in this function to make sure there is only one action, otherwise throw an error?

Reply:
How does this look like?
```
  /**
   * Retrieve an action on a specific task with its type.
   * Error will be thrown if more than one action is found.
   * @param name The task name.
   * @return The action pointer. Null means that such an action does not exist.
   */
  template <class T>
  const T * getActionByTask(const std::string & task)
  {
    const auto it = _action_blocks.find(task);
    if (it == _action_blocks.end())
      return nullptr;

    T * p = nullptr;
    for (const auto & action : it->second)
    {
      T * tp = dynamic_cast<T *>(action);
      if (tp)
      {
        if (p)
          mooseError("More than one actions have been detected in getActionByTask");
        else
          p = tp;
      }
    }
    return p;
  }
```